docs/Reference/swagger/swagger-ui/oauth2-redirect.html

```diff
@@ -0,0 +1,67 @@
+<!doctype html>
+<html lang="en-US">
+<body onload="run()">
+</body>
+</html>
+<script>
+    'use strict';
+    function run () {
+        var oauth2 = window.opener.swaggerUIRedirectOauth2;
+        var sentState = oauth2.state;
+        var redirectUrl = oauth2.redirectUrl;
+        var isValid, qp, arr;
+
+        if (/code|token|error/.test(window.location.hash)) {
+            qp = window.location.hash.substring(1);
+        } else {
+            qp = location.search.substring(1);
+        }
+
+        arr = qp.split("&")
+        arr.forEach(function (v,i,_arr) { _arr[i] = '"' + v.replace('=', '":"') + '"';})
+        qp = qp ? JSON.parse('{' + arr.join() + '}',
+                function (key, value) {
+                    return key === "" ? value : decodeURIComponent(value)
+                }
+        ) : {}
+
+        isValid = qp.state === sentState
+
+        if ((
+          oauth2.auth.schema.get("flow") === "accessCode"||
+          oauth2.auth.schema.get("flow") === "authorizationCode"
+        ) && !oauth2.auth.code) {
+            if (!isValid) {
+                oauth2.errCb({
+                    authId: oauth2.auth.name,
+                    source: "auth",
+                    level: "warning",
+                    message: "Authorization may be unsafe, passed state was changed in server Passed state wasn't returned from auth server"
```

Comment:
```suggestion
                    message: "Authorization might be unsafe, passed state was changed in server, but passed state wasn't returned from auth server"
```

Reply:
from my favorite resource, the [Google developer documentation style guide](https://developers.google.com/style/word-list?hl=en#letter-m) 
(At Tendermint, we often use "may" incorrectly, along with other habitual fun grammar twists)

**might**
Use _might_ to convey possibility or an uncertain outcome (for example, You might be prompted to enter your credentials). 
**may**
In general, reserve _may_ for official policy or legal considerations. To convey possibility, use _can_ or _might_ instead. To convey permission, use _can_ instead.